src/ray/object_manager/object_buffer_pool.cc

```diff
@@ -290,6 +320,16 @@ ray::Status ObjectBufferPool::EnsureBufferExists(const ObjectID &object_id,
 
 void ObjectBufferPool::FreeObjects(const std::vector<ObjectID> &object_ids) {
   absl::MutexLock lock(&pool_mutex_);
+  auto not_in_copy = [this, object_ids]() {
```

Comment:
Based on the logs of the failing test, what is likely happening is:
- An unguarded copy is in flight
- ObjectBufferPool::FreeObjects is called
- Which calls `store_client_->Delete` 
- Which internally calls `PlasmaAllocator::Free` making a call to `dlfree` which triggers a `SIGSEGV`